### PR TITLE
chore(backend): deploy to base sepolia

### LINKED ISSUE #16: backend: Deploy to Base Sepolia

### PR DESCRIPTION
**Pull request by jo-elimu**
closes #16

### CODE REVIEW

**Comment by jo-elimu** [comment]
@coderabbitai Why is `npx hardhat clean` failing with this error?:

```
HardhatError: HH8: There's one or more errors in your config file:

  * Invalid account: #0 for network: base_sepolia - Expected string, received undefined
  * Invalid account: #0 for network: base_mainnet - Expected string, received undefined
```

File: backend/hardhat.config.ts [hunk 1]
```diff
@@ -1,10 +1,48 @@
 import { HardhatUserConfig } from "hardhat/config";
 import "@nomicfoundation/hardhat-toolbox";
 
+require("dotenv").config();
```
Comment (Bender-22) [inline backend/hardhat.config.ts:4]:
```suggestion
require("dotenv").config();

const WALLET_KEY = process.env.WALLET_KEY || "";
```

File: backend/hardhat.config.ts [hunk 2]
```diff
@@ -1,10 +1,48 @@
...
+  solidity: {
+    version: "0.8.24"
+  },
   gasReporter: {
     enabled: true
+  },
+  networks: {
+    base_sepolia: { // Chain ID 84532
+      url: "https://sepolia.base.org",
+      accounts: [process.env.WALLET_KEY as string]
```
Comment (Bender-22) [inline backend/hardhat.config.ts:16]:
```suggestion
      accounts: [WALLET_KEY]
```

File: backend/hardhat.config.ts [hunk 3]
```diff
@@ -1,10 +1,48 @@
...
+    },
+    base_mainnet: { // Chain ID 8453
+      url: "https://mainnet.base.org",
+      accounts: [process.env.WALLET_KEY as string]
```
Comment (Bender-22) [inline backend/hardhat.config.ts:20]:
```suggestion
      accounts: [WALLET_KEY]
```

**Review by Bender-22 — DISMISSED** [review]

**Comment by jo-elimu** [comment]
> 1. **Environment Variables**: Ensure that the `WALLET_KEY` environment variable is correctly set in your environment or in a `.env` file. This variable should contain the private key for the wallet accounts used in the `base_sepolia` and `base_mainnet` networks.

@coderabbitai Thanks. Fixed it by using an empty array, `[]`, when the `WALLET_KEY` variable is missing.

**Review by Bender-22 — APPROVED** [review]

**Pull request merged**